app/controllers/steps_controller.rb

```diff
@@ -29,13 +25,31 @@ def step_attrs
     step_class.attribute_names
   end
 
+  def current_path(params = nil)
+    step_path(self.class, params)
+  end
+
+  def maybe_skip
+    if skip?
+      if going_backwards?
+        redirect_to previous_path(rel: "back")
+      else
+        redirect_to next_path
+      end
+    end
+  end
```

Comment:
Nested conditionals can sometimes feel a little smelly so what do you think about breaking this down to ...

```ruby
def maybe_skip
  if going_backwards?
    redirect_to previous_path(rel: "back")
  elsif skip?
    redirect_to next_path
  end
end

# ... ✂

def going_backwards?
  skip? && params["rel"] == "back"
end
```

I'm looking at this and haven't *quite* decided which approach is better. I'll leave it up to you to decide how you feel about it and will defer to your best judgement. Just presenting that as an option 😄